dask/tests/test_tokenize.py

```diff
@@ -31,54 +37,29 @@
 pd = import_or_none("pandas")
 
 
-def _clear_function_cache():
-    from dask.base import function_cache, function_cache_lock
-
-    with function_cache_lock:
-        function_cache.clear()
-
-
-def check_tokenize(*args, idempotent=True, deterministic=None, copy=None, **kwargs):
-    """Wrapper around tokenize
-
-    Parameters
-    ----------
-    args, kwargs: passed to tokenize
-    idempotent: True or False
-        If True, expect tokenize() called on the same object twice to produce the same
-        result. If False, expect different results. Default: True
-    deterministic: True, False, or "maybe"
-        If True, expect tokenize() called on two identical copies of an object to
-        produce the same result. If False, expect different results. If "maybe", expect
-        nothing. Default: same as idempotent
-    copy: callable or False
-        f(T)->T that deep-copies the object. Default: distributed serialize
-    """
-    if deterministic is None:
-        deterministic = idempotent
-
-    if copy is None:
-        copy = lambda x: cloudpickle.loads(cloudpickle.dumps(x))
-
-    ensure_deterministic = deterministic is True  # not maybe
-    with dask.config.set({"tokenize.ensure-deterministic": ensure_deterministic}):
+def check_tokenize(*args, **kwargs):
+    with dask.config.set({"tokenize.ensure-deterministic": True}):
         before = tokenize(*args, **kwargs)
 
         # Test idempotency (the same object tokenizes to the same value)
-        _clear_function_cache()
         after = tokenize(*args, **kwargs)
 
-        assert (before == after) is idempotent
+        assert before == after
 
-        # Test same-interpreter determinism (two identical objects tokenize to
-        # the same value as long as you do it on the same interpreter)
-        if copy:
-            args2, kwargs2 = copy((args, kwargs))
-            _clear_function_cache()
-            after = tokenize(*args2, **kwargs2)
+        # Test same-interpreter determinism (two identical objects tokenize to the
+        # same value as long as you do it on the same interpreter) We are not
+        # particularly interested in a class that's never been pickled vs. one
+        # that's already been pickled already (cloudpickle can introduce artifacts
+        # on the first round-trip). We do care however about classes that have both
+        # been through a serialization roundtrip at least once (not necessarily the
+        # same amount of times).
+        args2, kwargs2 = cloudpickle.loads(cloudpickle.dumps((args, kwargs)))
+        args3, kwargs3 = cloudpickle.loads(cloudpickle.dumps((args, kwargs)))
+        args3, kwargs3 = cloudpickle.loads(cloudpickle.dumps((args3, kwargs3)))
 
-            if deterministic != "maybe":
-                assert (before == after) is deterministic
+        tok2 = tokenize(*args2, **kwargs2)
+        tok3 = tokenize(*args3, **kwargs3)
+        assert tok2 == tok3
```

Comment:
I'm not sure if this is what you are referring to in the comment above but don't we want this to be equal as before?
```suggestion
        assert tok2 == tok3 == before
```

Reply:
This PR incorporates https://github.com/dask/dask/pull/10883.
To review this PR before the above is merged, just look at the last commit.
This comment is about a change in #10883; please keep the critique there.